lib/hash.cpp

```diff
@@ -1,139 +1,346 @@
+/*
+ *  xxHash - Fast Hash algorithm
+ *  Copyright (C) 2012-2021, Yann Collet
+ *
+ *  BSD 2-Clause License (http://www.opensource.org/licenses/bsd-license.php)
```

Comment:
BSD 2 is compatible with Apache, so this should not cause license problems.

Reply:
xxhash is also used in LLVM. LLVM is under Apache2 (+ LLVM exception), so there is a precedent just in case.